exporter/loggingexporter/factory.go

```diff
@@ -32,6 +32,16 @@ func NewFactory() exporter.Factory {
 	return exporter.NewFactory(
 		typeStr,
 		createDefaultConfig,
+		exporter.WithTraces(createTracesExporter, component.StabilityLevelDeprecated),
+		exporter.WithMetrics(createMetricsExporter, component.StabilityLevelDeprecated),
+		exporter.WithLogs(createLogsExporter, component.StabilityLevelDeprecated),
+	)
+}
+
+func NewFactoryWithName(s component.Type) exporter.Factory {
+	return exporter.NewFactory(
+		s,
+		createDefaultConfig,
```

Comment:
Instead of introducing the exporter `NewFactoryWithName` and keep it for a year, maybe we can move `createDefaultConfig` to `exporter/internal/` package and have both debug and logging exporters use it?

Reply:
Let me try that. I'll move this PR back to draft while I try this out.